src/app/editor-common/components/type-select/type-select.component.ts

```diff
@@ -120,12 +130,11 @@ export class InputTypeSelectComponent extends DirectiveBase implements ControlVa
         this.onTouched = fn;
     }
 
-    setDisabledState(isDisabled: boolean) {
-        if (isDisabled) {
-            this.skipOnChange = true;
-            this.form.controls["type"].disable();
+    setDisabledState(disabled: boolean) {
+        if (disabled === true) {
```

Comment:
How about:
const control = this.form.controls["isItemOrArray"]
disabled ? control.disable() : control.enable();